llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1874,6 +1874,131 @@ LogicalResult PackOp::generateScalarImplementation(OpBuilder &builder,
   return success();
 }
 
+SmallVector<Operation *>
+PackOp::getTiledImplementation(OpBuilder &builder,
+                               ArrayRef<OpFoldResult> offsets,
+                               ArrayRef<OpFoldResult> sizes) {
+  Location loc = getLoc();
+  auto ctx = builder.getContext();
+
+  // Take the minimum of two integers.
+  auto idMap = AffineMap::getMultiDimIdentityMap(2, ctx);
+  auto min = [&](OpFoldResult v1, OpFoldResult v2) -> OpFoldResult {
+    return builder.createOrFold<AffineMinOp>(
+        loc, idMap,
+        ValueRange{getValueOrCreateConstantIndexOp(builder, loc, v1),
+                   getValueOrCreateConstantIndexOp(builder, loc, v2)});
+  };
+  // Subtract two integers.
+  AffineExpr dim0, dim1;
+  bindDims(ctx, dim0, dim1);
+  auto subMap = AffineMap::get(2, 0, {dim0 - dim1});
+  auto sub = [&](OpFoldResult v1, OpFoldResult v2) -> OpFoldResult {
```

Comment:
I think now you can use `makeComposedFoldedAffineApplyOp` . That probably gives you what you want.